streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsProducer.java

```diff
@@ -52,17 +55,19 @@
     private final Logger log;
     private final String logPrefix;
 
-    private final Producer<byte[], byte[]> producer;
+    private Producer<byte[], byte[]> producer;
+    private final StreamThread.ProcessingMode processingMode;
```

Comment:
Why we need to keep both processingMode and eosEnabled?

Reply:
It's simpler IMHO -- for most duties like "initTx" or "beginTx" we don't care if its eos-alpha or eos-beta. If we only keep `processingMode` we need to rewrite:
```
if (eosEnabled) {
...
}
```
to
```
if (processingMode == EXACTLY_ONCE_ALPHA || processingMode == EXACTLY_ONCE_BETA) {
...
}
```

what seem to be more "clumsy" ?

Reply:
My concern is that keep two variables is always more vulnerable to bugs where the two mistakenly have different values (e.g. eosEnabled == false while mode == EXACTLY_ONCE). Instead, we can have a private method `eosEabled()` which is based on the mode variable and then be called whenever possible.

Reply:
Works for me